scripts/test-matrix.mjs

```diff
@@ -4,124 +4,79 @@
-    ],
-    {
-      stdio: ["ignore", "pipe", "inherit"],
-      env: { TEST_ARGS: "ios" },
-      encoding: "utf-8",
-    }
-  );
-  if (status !== 0) {
-    throw new Error(
-      "An error occurred while trying to evaluate 'wdio.config.mjs'"
-    );
-  }
-  return stdout.trim();
-});
-
-function log(message = "", tag = TAG) {
-  console.log(tag, message);
-}
-
-/**
- * Invokes `react-native run-<platform>`.
- * @param {Platform} platform
- */
-function buildAndRun(platform) {
-  switch (platform) {
-    case "ios": {
-      const simulator = getIOSSimulatorName();
-      $(PACKAGE_MANAGER, platform, "--simulator", simulator, "--no-packager");
-      break;
-    }
-    default: {
-      $(PACKAGE_MANAGER, platform, "--no-packager");
-      break;
-    }
-  }
-}
-
-/**
- * Configures app.
- * @param {Platform} platform
- * @param {{ hermes?: boolean; newArch?: boolean; }} config
- */
-function configure(platform, { hermes, newArch }) {
-  switch (platform) {
-    case "android":
-      // Hermes is always enabled on Android
-      if (newArch) {
+/** @type {Record<TargetPlatform, PlatformConfig>} */
+const PLATFORM_CONFIG = {
+  android: {
+    name: "Android",
+    engines: ["hermes"],
+    configure: ({ variant }) => {
+      if (variant === "fabric") {
         const properties = "android/gradle.properties";
         const content = readTextFile(properties);
         fs.writeFileSync(
           properties,
           content.replace("#newArchEnabled=true", "newArchEnabled=true")
         );
       }
-      break;
-    case "ios": {
-      const podfile = `${platform}/Podfile`;
-      let content = readTextFile(podfile);
-      if (hermes) {
-        content = content.replace(
-          ":hermes_enabled => false",
-          ":hermes_enabled => true"
-        );
-      }
-      if (newArch) {
-        content = content.replace(
-          ":fabric_enabled => false",
-          ":fabric_enabled => true"
-        );
-      }
-      fs.writeFileSync(podfile, content);
-      break;
-    }
-  }
-}
 
-/**
- * Invokes `pod install` for specified platform.
- * @param {ApplePlatform} platform
- */
-function installPods(platform) {
-  const options = {
-    force: true,
-    maxRetries: 3,
-    recursive: true,
-    retryDelay: 500,
-  };
-  fs.rmSync(`${platform}/Podfile.lock`, options);
-  fs.rmSync(`${platform}/Pods`, options);
-  fs.rmSync(`${platform}/build`, options);
-  $("pod", "install", `--project-directory=${platform}`);
+      return true;
+    },
+  },
+  ios: {
+    name: "iOS",
+    engines: ["jsc", "hermes"],
+    configure: configurePodfile,
+    prebuild: installPods,
+  },
+  macos: {
+    name: "macOS",
+    engines: ["jsc", "hermes"],
+    configure: configurePodfile,
+    prebuild: installPods,
+  },
+  visionos: {
+    name: "visionOS",
+    engines: ["jsc", "hermes"],
+    configure: configurePodfile,
+    prebuild: installPods,
+  },
+  windows: {
+    name: "Windows",
+    engines: ["hermes"],
```

Comment:
I guess, no v8?

Reply:
Is the V8 path even public?

Reply:
@acoates-ms ?

Reply:
For now though, let's keep it like this. Windows will be enabled in the next PR. We can consider V8 then.

Reply:
We dont provide support for the V8 path - I wouldn't bother including it.